PopPUNK/models.py

```diff
@@ -1006,34 +1027,50 @@ class LineageFit(ClusterFit):
             The ranks used in the fit
     '''
 
-    def __init__(self, outPrefix, ranks, use_gpu = False):
+    def __init__(self, outPrefix, ranks, max_search_depth, reciprocal_only, count_unique_distances, dist_col = None, use_gpu = False):
         ClusterFit.__init__(self, outPrefix)
         self.type = 'lineage'
         self.preprocess = False
+        self.max_search_depth = max_search_depth
+        self.nn_dists = None # stores the unprocessed kNN at the maximum search depth
         self.ranks = []
         for rank in sorted(ranks):
             if (rank < 1):
                 sys.stderr.write("Rank must be at least 1")
                 sys.exit(0)
             else:
                 self.ranks.append(int(rank))
+        self.lower_rank_dists = {}
```

Comment:
Necessary?

Reply:
Possibly not, but makes the object's structure clearer to me (contrasts with list of ranks)